State/ApplicationState.cpp

```diff
@@ -125,4 +130,76 @@ moonlight_xbox_dx::ApplicationState^ __stateInstance;
 moonlight_xbox_dx::ApplicationState^ moonlight_xbox_dx::GetApplicationState() {
 	if (__stateInstance == nullptr)__stateInstance = ref new moonlight_xbox_dx::ApplicationState();
 	return __stateInstance;
+}
+
+void moonlight_xbox_dx::ApplicationState::WakeHost(MoonlightHost^ host)
+{
+	if (host == nullptr || host->Connected) return;
+
+	if (host->MacAddress == nullptr || host->MacAddress == "00:00:00:00:00:00") {
+		Utils::Log("No recorded Mac address, the client and host must be connected at least once.\n");
+		return;
+	}
+
+	WSADATA wsa;
+	if (WSAStartup(MAKEWORD(2, 2), &wsa) != 0) {
+		Utils::Log("Websocket startup failed.\n");
+		return;
+	}
+
+	SOCKET s = socket(AF_INET, SOCK_DGRAM, IPPROTO_UDP);
+	if (s == SOCKET_ERROR || s == INVALID_SOCKET) {
+		Utils::Log("Websocket creation failed.\n");
+		WSACleanup();
+		return;
+	}
+	
+	std::string hostIP = Utils::PlatformStringToStdString(host->LastHostname).substr(0, 14);
```

Comment:
We might not be sure that LastHostname is a correct IP address. It might be a domain name or the local computer name. Only mDNS adds the machine via the IP address